src/update_engine/delta_performer.cc

```diff
@@ -461,10 +464,18 @@ bool DeltaPerformer::PerformReplaceOperation(
   // point to one of the ExtentWriter objects above.
   ExtentWriter* writer = NULL;
   if (operation.type() == DeltaArchiveManifest_InstallOperation_Type_REPLACE) {
-    writer = &zero_pad_writer;
+    if (is_kernel_partition) {
```

Comment:
That's a pretty fussy error message. I don't have an opinion one way or the other but don't bother considering "upstream" when picking what to do, we are solidly forked at this point. Chromium can be useful as a reference at times but that's as far as it goes.